drivers/cc110x/cc110x_netdev.c

```diff
@@ -315,7 +328,7 @@ static int cc110x_init(netdev_t *netdev)
      */
     dev->addr = dev->params.l2addr;
     while (dev->addr == CC110X_L2ADDR_AUTO) {
-        luid_get(&dev->addr, 1);
+        _cc110x_eui_get(netdev, &dev->addr);
     }
```

Comment:
I have the feeling that this loop could cause issues, if the user provided `cc110x_eui_get()` would provide `CC110X_L2ADDR_AUTO` as EUI. I think just having an API contract that this address must not be returned and an `assert()` here to enforce this would solve the issue.

How about making `cc110x_eui_get()` return void and calling it here without the `while`? The `CC110X_L2ADDR_AUTO` mechanism was only a crude way to provide an L2 address via compile time flags. Since you are providing a proper mechanism in this PR, IMO we can just drop this macro and the `l2addr` member of `cc110x_params_t`. Also the documentation in `drivers/include/cc110x.h` lines 85 to 94 needs update - IMO just reference the `cc110x_eui_get` function there.

The default `weak` `cc110x_eui_get()` should loop until the returned byte differs from `CC1XXX_BCAST_ADDR` to match the API contract. (Actually, the loop is not needed, as the luid_get would do a pretty poor job in returning the same ID twice in two subsequent calls. But a `do`...`while` loop there should be more ROM efficient and is bullet proof, just in case a poor `luid_get()` gets used or the `init()` gets interrupted by someone doing excessive calls to `luid_get()`.)

Btw.: It might make sense to place the default `weak` implementation in `drivers/cc1xxx_common` and call the provider `cc1xxx_eui_get()`. Just in case a [cc1020 driver](https://github.com/RIOT-OS/RIOT/issues/658) magically gets upstream, this would be useful by both drivers.

Reply:
So this would be ok?
```C
void __attribute__((weak)) cc1xxx_eui_get(const netdev *netdev, void *eui)
{
    do {
        luid_get(eui, 1);
    }
    while (*((uint8_t *)eui) == CC1XXX_BCAST_ADDR);
}
``` 
And it should be defined in `gnrc_netif_cc1xxx.c`?

Reply:
Exactly. But since all transceivers of the CC1xxx family have `uint8_t` as address type, you could also just use `uint8_t *` instead of `void *`.